publication/models/account_analytic_invoice_line.py

```diff
@@ -28,3 +28,55 @@ def action_distribution_list(self):
         action['view_mode'] = 'form'
         action['target'] = 'current'
         return action
+
+    @api.model
+    def create(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).create(vals)
+        if result.publication:
+            self._create_publication_list()
+        return result
+
+    @api.multi
+    def write(self, vals):
+        result = super(AccountAnalyticInvoiceLine, self).write(vals)
+        self.filtered(
+            lambda x: x.publication is True)._create_publication_list()
+        return result
+
+    @api.multi
+    def _create_publication_list(self):
+        """ Conditionally creates a publication list:
+        If a list with the same contract_partner_id, product_id and
+        partner_id exists go write on that, else create a new one.
+        """
+        d_list = self.env['distribution.list']
+        for rec in self:
+            pub_list = d_list.search([
```

Comment:
partner and contract partner are the same on the distribution list entries that are automatically created. To keep the code simpel add an assignment ```partner = rec.analytic_account_id.partner_id``` and use this for the selection both of partner_id and contract_partner_id